### PR TITLE
feat: add viewTransition for dark mode

### PR DESCRIPTION
**Pull request by fwqaaq**
### <samp>🤖 Generated by Copilot at 2d3ba6a</samp>

This pull request adds a circular clip-path animation for the dark mode transition in Chromium browsers that support the `::view-transition` pseudo-element. It also improves the interactivity and responsiveness of the dark mode switch and the home page list items. It refactors the `useDarkMode` hook and its dependencies, and adds a utility function for feature detection. It updates the CSS variables and rules to avoid animation conflicts.

### <samp>🤖 Generated by Copilot at 2d3ba6a</samp>

> _In the dark mode we rise, we switch the light with a click_
> _We use the `::view-transition` to animate the trick_
> _But we check the browser and the motion preference_
> _We don't want to break the page or cause any interference_

### <samp>🤖 Generated by Copilot at 2d3ba6a</samp>

*  Add dark mode transition feature using circular clip-path animation and `::view-transition` pseudo-element ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-a5cebf01f5a3695ff1f69ee076704d693ad21263bdfa158ed24867e8beba2bd9L13-R14), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-159f736c91e4f8289751d620dff31bc77c7c62fea4c5a110f867777f4078ce37L29-R37), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L1-R6), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L12-R12), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L27-R29), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L38-R50), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L79-R75), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L117-R148), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L142-R165), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0R185-R186), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-da10bcc7d8dff796f50c9358a7268eb5abd2c50f460aecc4b2be1c691a966224R29-R33))
  * Import `MouseEvent` type from React and `isViewTransition` function from `utils.ts` in `useDarkMode.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L1-R6))
  * Add `transition` option to `DarkModeConfig` interface and `IMediaStore` interface in `useDarkMode.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L12-R12), [link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L27-R29))
  * Add `mousePosition` state and `setMousePosition` function to `useDarkModeInternal` hook in `useDarkMode.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L38-R50))
  * Simplify logic for getting and setting dark mode state from storage in `useDarkModeInternal` hook in `useDarkMode.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L79-R75))
  * Add `setDarkModeClass` function and `transition` logic to `useDarkModeInternal` hook in `useDarkMode.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L117-R148))
  * Pass event object to `setMousePosition` and `toggle` functions in `useDarkModeInternal` hook in `useDarkMode.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0L142-R165))
  * Pass `transition` option to `useDarkModeInternal` hook in `useDarkMode` hook in `useDarkMode.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-2047312bd1019e8eb8a5db53adac4005b498148357ba5184c72fb95e071a5ed0R185-R186))
  * Replace `onChange` prop with `onClick` prop and pass event object to `toggle` function in `DarkModeSwitch.tsx` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-a5cebf01f5a3695ff1f69ee076704d693ad21263bdfa158ed24867e8beba2bd9L13-R14))
  * Add CSS rule to disable animation for `::view-transition-old(root)` and `::view-transition-new(root)` pseudo-elements in `variables.css` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-159f736c91e4f8289751d620dff31bc77c7c62fea4c5a110f867777f4078ce37L29-R37))
  * Add `isViewTransition` function to `utils.ts` to check browser support and user preference for `::view-transition` pseudo-element ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-da10bcc7d8dff796f50c9358a7268eb5abd2c50f460aecc4b2be1c691a966224R29-R33))
* Add `cursor-pointer` class to list items displaying subfeatures of the app in `page.tsx` ([link](https://github.com/Crossbell-Box/xLog/pull/518/files?diff=unified&w=0#diff-083b47e335af1cbf2a7f861ccfce83cb51b0c1ca359cc4d7b68490ea9ffee7a6L217-R217))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

0x4eECd2da8E95cb559291336AE6dadfB96A76B6A9

### CODE REVIEW

**Comment by fwqaaq** [comment]
Hi, sorry, I'm not very familier with `React`. I use newr [View Transition API](https://developer.mozilla.org/en-US/docs/Web/API/ViewTransition) to animate the interaction between `dark` and `light`. And some others modifications.

File: src/lib/utils.ts [hunk 1]
```diff
@@ -26,6 +26,11 @@ export function pick<T, K extends keyof T>(obj: T, keys: K[]): Pick<T, K> {
 
 export const isServerSide = () => typeof window === "undefined"
 
+export const isViewTransition = () =>
+  // @ts-ignore
+  !!document.startViewTransition &&
```
Comment (DIYgod) [inline src/lib/utils.ts:31]: This function will throw an error on the server side, so we need to use isServerSide to exclude server-side execution.
image

**Review by DIYgod — CHANGES_REQUESTED** [review]

File: src/lib/utils.ts (hunk 1 above)
Comment (fwqaaq) [reply src/lib/utils.ts:31]: @DIYgod Thanks ❤️ !  And I added `stack context` to make the animation look more natural,

**Comment by DIYgod** [comment]
Thanks so much!

**Comment by fwqaaq** [comment]
> Thanks so much!

Thanks DIYgod, I'm your fan ❤️ .

**Pull request merged**